### PR TITLE
[11.0][ADD] stock_request_tier_validation

### PR DESCRIPTION
**Pull request by HviorForgeFlow**
This module extends the functionality of Stock Requests and Stock Request Orders to support a tier validation process.

A default tier is created allowing Stock Request Manager to approve Stock Request and Stock Request Orders.

In addition, you may want to add more tiers, so:

#. Go to *Settings > Technical > Tier Validations > Tier Definition*.
#. Create as many tiers as you want for Purchase Order model.

To use this module, you need to:

#. Create a Stock Request triggering at least one "Tier Definition".
#. Click on *Request Validation* button.
#. Under the tab *Reviews* have a look to pending reviews and their statuses.
#. Once all reviews are validated button to *Confirm* will be shown.

### CODE REVIEW

**Review by LoisRForgeFlow — APPROVED** [review]
LGTM :+1: 

Can you squash?

**Comment by HviorForgeFlow** [comment]
done @lreficent

**Comment by JordiBForgeFlow** [comment]
@hveficent When there's no active tier validation to apply (or all are inactive) you are stuck. There's no button "Confirm" available. Just Cancel.

**Comment by HviorForgeFlow** [comment]
@jbeficent good catch, fixed, thanks

**Review by JordiBForgeFlow — APPROVED** [review]

**Comment by rousseldenis** [comment]
@jbeficent say when it's ok

File: stock_request_tier_validation/data/stock_request_tier_definition.xml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<!-- Copyright 2019 Eficent Business and IT Consulting Services S.L.
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo>
```
Comment (Nikul-Chaudhary) [inline stock_request_tier_validation/data/stock_request_tier_definition.xml:4]: `<odoo noupdate="1">`

**Review by Nikul-Chaudhary — COMMENTED** [review]
Improve Code

**Review by rousseldenis — APPROVED** [review]
Code review

**Pull request merged**